### PR TITLE
nc-news-5

### PR DESCRIPTION
**Pull request by SergMo**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -84,4 +84,41 @@ describe('GET /api/articles/:article_id', () => {
...
+});
+
+describe('GET /api/articles', () => {
+	test('GET:200 should return an array of articles', () => {
+		return request(app)
+			.get('/api/articles')
+			.expect(200)
+			.then((response) => {
+				const articles = response.body.articles;
+				console.log(articles[1]);
```
Comment (Keb84) [inline __tests__/app.test.js]: Please remove any console logs before pushing your work.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -84,4 +84,41 @@ describe('GET /api/articles/:article_id', () => {
...
+				expect(articles).toEqual(expect.any(Array));
+				expect(articles.length).toBeGreaterThan(0);
```
Comment (Keb84) [inline __tests__/app.test.js]: As you can see the data and how many items are in your array of test data you could be more specific by asserting that (articles.length).toBe(the actual length).

File: __tests__/app.test.js [hunk 3]
```diff
@@ -84,4 +84,41 @@ describe('GET /api/articles/:article_id', () => {
...
+				articles.forEach((article) => {
+					expect(article).toHaveProperty('author');
+					expect(article).toHaveProperty('title');
+					expect(article).toHaveProperty('article_id');
+					expect(article).toHaveProperty('topic');
+					expect(article).toHaveProperty('created_at');
+					expect(article).toHaveProperty('votes');
+					expect(article).toHaveProperty('article_img_url');
+					expect(article).toHaveProperty('comment_count');
+					expect(article).not.toHaveProperty('body');
```
Comment (Keb84) [inline __tests__/app.test.js]: By using a jest matcher like toMatchObject you could check all the properties in the object without repeating yourself with toHaveProperty each time. 

You could also add the matcher expect.any() to check the values are strings or numbers for example.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -84,4 +84,41 @@ describe('GET /api/articles/:article_id', () => {
...
+	});
+	test('articles should be sorted by date in descending order', () => {
+		return request(app)
+			.get('/api/articles')
+			.expect(200)
+			.then((response) => {
+				const articles = response.body.articles;
+				const dates = articles.map((article) => article.created_at);
+				const sortDates = [...dates].sort((a, b) => new Date(b) - new Date(a));
+				expect(dates).toEqual(sortDates);
```
Comment (Keb84) [inline __tests__/app.test.js]: This is a nice test 👍 . There is a great jest matcher called toBeSotedBy that can be installed for this test. It makes testing for sorted by ascending or descending a little clearer in your code. Here is an example 
```.toBeSortedBy('count', {descending: true})```

File: app.js [hunk 5]
```diff
@@ -2,13 +2,14 @@ const express = require('express');
 const app = express();
 const { getAllTopics } = require('./controllers/topic.controller');
 const { getApiDocumentation } = require('./controllers/api.controller');
-const { getArticleById } = require('./controllers/article.controller');
+const { getArticleById, getAllArticles } = require('./controllers/article.controller');
 app.use(express.json());
```
Comment (Keb84) [inline app.js]: You shouldn't need express.json in here yet as you only have GET requests.

File: models/article.model.js [hunk 6]
```diff
@@ -9,4 +9,16 @@ exports.selectArticleById = (article_id) => {
...
+	return db
+		.query('SELECT article_id, title, author, topic, created_at, votes, article_img_url FROM articles ORDER BY created_at DESC')
+		.then((result) => result.rows);
+}
+
+exports.countArticleComments = (article_id) => {
+	return db
+		.query('SELECT COUNT(*) FROM comments WHERE article_id = $1', [article_id])
+		.then((result) => parseInt(result.rows[0].count))
 }
```
Comment (Keb84) [inline models/article.model.js]: We want to try and reduce the amount of calls to the database. Consider a way you could do this with one db.query to get all the rows including the comment count? Have a look at aggregate functions in the sql relationships section of nc notes 🙂

**Review by Keb84 — APPROVED** [review]
Really nice work on this ticket 🤸 please have a look at my comments and merge when ready 🚀

File: __tests__/app.test.js (hunk 2 above)
Comment (SergMo) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 1 above)
Comment (SergMo) [reply __tests__/app.test.js]: done

**Review by Keb84 — APPROVED** [review]
Nicely done. Merge when you are ready 🙂

File: models/article.model.js (hunk 6 above)
Comment (SergMo) [reply models/article.model.js]: done

File: __tests__/app.test.js (hunk 3 above)
Comment (SergMo) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 4 above)
Comment (SergMo) [reply __tests__/app.test.js]: done

**Pull request merged**